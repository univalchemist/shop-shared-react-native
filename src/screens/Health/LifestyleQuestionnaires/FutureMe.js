import React, { useState, useEffect, useRef } from 'react';
import PropTypes from 'prop-types';
import {
  Container,
  Image,
  Flex,
  Box,
  Text,
  UploadBox,
} from '@wrappers/components';
import { getImageFromImagePicker } from '@utils/nativeImageHelpers';
import { withTheme } from 'styled-components/native';
import { healthFutureMeImg } from '@images';
import { compose } from 'redux';
import { FormattedMessage } from 'react-intl';
import { HEALTH_PHOTO_VIEWER_MODAL } from '@routes';
import { useIntl } from '@wrappers/core/hooks';
import { logAction, categories } from '@store/analytics/trackingActions';
import styled from 'styled-components/native';
import theme from '@theme';
import { NAVIGATOR_HEIGHT } from '@utils';
import { connect } from 'react-redux';

const HintText = styled.Text`
  margin: 6px 0 10px;
  font-size: 12px;
  color: ${theme.colors.gray[1]};
`;

const enhance = compose(
  withTheme,
  connect(({ health: { faceAging } }) => ({ faceAging })),
);

const FutureMe = ({
  change,
  initialValues,
  navigation,
  uploadBoxLayout,
  faceAging,
}) => {
  const intl = useIntl();
  const uploadBoxRef = useRef();
  useEffect(() => {
    if (initialValues && initialValues.image) setDocument(initialValues.image);
    else setDocument(null);

    setIsError(faceAging.isError);
  }, []);
  useEffect(() => {
    setIsError(faceAging.isError);
  }, [faceAging]);

  const [document, setDocument] = useState(null);
  const [isError, setIsError] = useState(false);
  const [disableUpload, setDisableUpload] = useState(false);

  const openGallery = () => {
    setDisableUpload(true);
    getImageFromImagePicker({
      intl,
      onSuccess: image => {
        setDisableUpload(false);
        setIsError(false);
        setDocument(image);
        change(`image`, image);
        logAction({
          category: categories.HEALTH_QUESTION,
          action: 'Upload picture succesfully',
        });
      },
      onError: () => setDisableUpload(false),
      onCancel: () => setDisableUpload(false),
    });
  };

  const viewPhoto = () => {
    const { uri } = document;
    navigation.navigate(HEALTH_PHOTO_VIEWER_MODAL, {
      uri,
      onRemove: () => {
        change(`image`, null);
        setDocument(null);
      },
    });
  };

  const renderUploadBox = () => {
    return (
      <UploadBox
        disabled={disableUpload}
        width="100%"
        height="350"
        data={document && !isError ? { uri: document.uri } : null}
        onAdd={openGallery}
        onView={viewPhoto}
        accessibilityDocumentType={intl.formatMessage({
          id: 'uploadBox.accessibilityLabel.type.photo',
        })}
        accessibilityField={intl.formatMessage({
          id: 'futureMeTitle',
        })}
      />
    );
  };

  return (
    <Container>
      <Flex alignItems="center">
        <Box mt={3}>
          <Image
            width={148}
            height={161}
            source={healthFutureMeImg}
            resizeMode="contain"
          />
        </Box>
        <Box mt={3}>
          <Text
            lineHeight={37}
            fontSize={32}
            fontWeight={300}
            color="gray.0"
            paddingBottom={24}
            textAlign="center"
          >
            <FormattedMessage id="futureMeTitle" />
          </Text>
        </Box>

        <Box mt={3} width="100%">
          <Text lineHeight={22} fontSize={16} color="gray.0" textAlign="left">
            {<FormattedMessage id="futureMeHint" />}
          </Text>

          <Box>
            <HintText>{intl.formatMessage({ id: 'optional' })}</HintText>
          </Box>
        </Box>

        <Box
          ref={uploadBoxRef}
          onLayout={() => {
            uploadBoxRef.current.measure((x, y, w, h, px, py) => {
              uploadBoxLayout(py - NAVIGATOR_HEIGHT);
            });
          }}
          width="100%"
        >
          {renderUploadBox()}
        </Box>
      </Flex>
    </Container>
  );
};

FutureMe.propTypes = {
  change: PropTypes.func,
  initialValues: PropTypes.shape({
    image: PropTypes.object,
  }),
};
export default enhance(FutureMe);
