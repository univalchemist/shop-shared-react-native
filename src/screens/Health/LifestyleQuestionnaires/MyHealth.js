import React, { useCallback } from 'react';
import { Image, Text, Box, Container, Flex } from '@wrappers/components';
import { FormattedMessage } from 'react-intl';
import { myHealthImage } from '@images';
import { useTheme } from '@wrappers/core/hooks';
import { categories, logAction } from '@store/analytics/trackingActions';
import QuestionWithRadioButtons from './QuestionWithRadioButtons';

const questionTrackingActions = {
  hereditaryDiabetes: 'Parents/siblings diabetes question',
  uninterestFrequency: 'Depressed question',
};

export const MyHealth = ({ questions, submitCount }) => {
  const theme = useTheme();
  const trackingQuestion = useCallback(questionName => {
    questionTrackingActions[questionName] &&
      logAction({
        category: categories.HEALTH_QUESTION,
        action: questionTrackingActions[questionName],
      });
  }, []);

  return (
    <Container>
      <Flex justifyContent="center" flexDirection="row">
        <Image
          source={myHealthImage}
          width={148}
          height={161}
          resizeMode="contain"
        />
      </Flex>

      <Box mt={24}>
        <Text
          lineHeight={37}
          fontSize={32}
          fontWeight={300}
          color={theme.colors.gray[0]}
          paddingBottom={24}
          textAlign="center"
        >
          <FormattedMessage id="myHealthTitle" />
        </Text>
      </Box>

      {questions.map(question => (
        <Box key={question.questionId} mb={48}>
          <QuestionWithRadioButtons
            submitCount={submitCount}
            question={question}
            onChange={() => {
              trackingQuestion(question.name);
            }}
          />
        </Box>
      ))}
    </Container>
  );
};

export default MyHealth;
