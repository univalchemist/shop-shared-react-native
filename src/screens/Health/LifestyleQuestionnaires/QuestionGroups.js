export const CHOICES = 'choices';
export const HEALTH = 'health';
