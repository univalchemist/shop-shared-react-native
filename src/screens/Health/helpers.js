import { flatten } from 'ramda';

export const getInitialValuesFromQuestions = questions => {
  const reducer = (accumulator, question) => {
    accumulator[question.name] = question.defaultValue;
    return accumulator;
  };

  return Object.keys(questions.questionGroups)
    .map(questionGroupName => questions.questionGroups[questionGroupName])
    .reduce((accumulator, questions) => [...accumulator, ...questions], [])
    .reduce(reducer, {});
};

const flattenQuestionGroups = questions =>
  Object.keys(questions.questionGroups)
    .map(key => questions.questionGroups[key])
    .reduce((accumulator, questions) => [...accumulator, ...questions], []);

export const filterFormValues = (values, questions) => {
  const findParentQuestion = (dependentQuestion, questionArray) => {
    return questionArray.find(
      q => q.questionId === dependentQuestion.parentQuestionId,
    );
  };
  const isActive = (parentQuestion, dependentQuestion) => {
    return flatten([values[parentQuestion.name]]).some(v =>
      dependentQuestion.activationAnswer.includes(v),
    );
  };
  const questionArray = flattenQuestionGroups(questions);
  questionArray.forEach(question => {
    if (
      question.dependentQuestion &&
      !isActive(findParentQuestion(question, questionArray), question)
    ) {
      values[question.name] = null;
    }
  });
  return values;
};

export const assignDefaultValueForDependentQuestions = (values, questions) => {
  const questionArray = flattenQuestionGroups(questions);
  questionArray.forEach(question => {
    if (
      question.dependentQuestion &&
      (values[question.name] === null || values[question.name] === undefined)
    ) {
      values[question.name] = question.defaultValue;
    }
  });
  return values;
};
