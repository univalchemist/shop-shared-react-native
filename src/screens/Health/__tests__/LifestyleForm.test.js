import {
  Button,
  Loader,
  ScrollViewForStickyButton,
} from '@wrappers/components';
import { HEALTH } from '@routes';
import { renderForTest } from '@testUtils';
import React from 'react';
import { Alert } from 'react-native';
import { fireEvent, flushMicrotasksQueue } from 'react-native-testing-library';
import LifestyleForm, {
  LifestyleForm as PureLifestyleForm,
} from '../LifestyleForm';
import messages from '@messages/en-HK.json';
import { MyChoices } from '../LifestyleQuestionnaires/MyChoices';
import { MyHealth } from '../LifestyleQuestionnaires/MyHealth';
import english_questions from '../LifestyleQuestionnaires/questions/en_questions';
import * as QuestionGroups from '../LifestyleQuestionnaires/QuestionGroups';
import { getInitialValuesFromQuestions } from '../helpers';
import { submitUserLifestyleResponse } from '@store/health/actions';
import FutureMe from '../LifestyleQuestionnaires/FutureMe';

jest.mock('@utils/nativeImageHelpers', () => ({
  compressImage: jest.fn(x => x),
}));

const ethnicityOptions = {
  option1: 'option1',
  option2: 'option2',
};
jest.useFakeTimers();
jest.mock('react-native/Libraries/Alert/Alert', () => ({ alert: jest.fn() }));

jest.mock('@store/health/actions', () => ({
  submitUserLifestyleResponse: jest.fn(values => {
    if (values.height === 2) {
      const error = {
        response: { data: { message: 'Error' } },
      };
      throw error;
    }
    return {
      type: 'TEST_CALL',
      value: { value: {} },
    };
  }),
  fetchUserLifestyleResponse: jest.fn(() => {
    return {
      type: 'TEST_CALL',
      value: { value: {} },
    };
  }),
  downloadFaceAgingPhoto: jest.fn(() => {
    return {
      type: 'TEST_CALL',
      value: { value: {} },
    };
  }),
  resetLoader: jest.fn(() => {
    return {
      type: 'TEST_CALL',
      value: { value: {} },
    };
  }),
}));

const stateWithError = {
  form: {
    lifestyleForm: {
      values: {
        height: 2, // triggers error
        temptingFoodItems: [],
      },
    },
  },
  health: {
    data: {},
    faceAging: { image: { uri: '' } },
    ethnicityOptions,
    fetchFaceAgingImageCompleted: true,
    fetchHealthResponseCompleted: true,
    fetchUserLifestyleResponseCompleted: true,
    submittingUserLifestyleResponse: false,
  },
  user: {
    profile: { preferredLocale: 'en-HK' },
  },
};

const stateWithoutError = {
  form: {
    lifestyleForm: {
      values: { temptingFoodItems: [] },
    },
  },
  health: {
    data: {
      height: 1,
      weight: 1,
      waistCircumference: 1,
      ethnicity: 'ethnicity1',
    },
    faceAging: { image: { uri: '' } },
    ethnicityOptions,
    fetchFaceAgingImageCompleted: true,
    fetchUserLifestyleResponseCompleted: true,
    submittingUserLifestyleResponse: false,
  },
  user: {
    profile: { preferredLocale: 'en-HK' },
  },
};

const stateWithSubmitResponseLoader = {
  form: {
    lifestyleForm: {
      values: { temptingFoodItems: [] },
    },
  },
  health: {
    data: {},
    faceAging: { image: { uri: '' } },
    ethnicityOptions,
    fetchFaceAgingImageCompleted: true,
    fetchHealthResponseCompleted: true,
    fetchUserLifestyleResponseCompleted: true,
    submittingUserLifestyleResponse: true,
  },
  user: {
    profile: { preferredLocale: 'en-HK' },
  },
};

describe('LifestyleForm', () => {
  const renderLifestyleForm = ({ initialState = stateWithoutError }) => {
    return renderForTest(
      <LifestyleForm
        route={{
          params: { scrollToFutureMe: false },
        }}
      />,
      {
        initialState,
      },
    );
  };

  it('should have the correct labels', () => {
    const { getByText } = renderForTest(
      <LifestyleForm
        route={{
          params: { scrollToFutureMe: false },
        }}
      />,
      {
        initialState: stateWithoutError,
      },
    );
    [messages.height, messages.weight, messages.waistCircumference].forEach(
      label => {
        const element = getByText(label);
        expect(element).toBeDefined();
      },
    );
  });

  it('should show the lifestyle form page in english', () => {
    const initialStateWithFormRendered = {
      form: {
        lifestyleForm: {
          values: {
            temptingFoodItems: [],
          },
        },
      },
      health: {
        data: {
          height: 1,
          weight: 1,
          waistCircumference: 1,
          ethnicity: 'ethnicity1',
        },
        faceAging: { image: { uri: '' } },
        ethnicityOptions,
        fetchUserLifestyleResponseCompleted: true,
        submittingUserLifestyleResponse: false,
      },
      user: {
        profile: { preferredLocale: 'en-HK' },
      },
    };
    const lifestyleForm = renderForTest(
      <LifestyleForm
        route={{
          params: { scrollToFutureMe: false },
        }}
      />,
      {
        initialState: initialStateWithFormRendered,
      },
    ).toJSON();

    expect(lifestyleForm).toMatchSnapshot();
  });

  it('should show the lifestyle form page in chinese', () => {
    const initialStateWithFormRendered = {
      form: {
        lifestyleForm: {
          values: { temptingFoodItems: [] },
        },
      },
      health: {
        data: {
          height: 1,
          weight: 1,
          waistCircumference: 1,
          ethnicity: 'ethnicity1',
        },
        faceAging: { image: { uri: '' } },
        ethnicityOptions,
        fetchUserLifestyleResponseCompleted: true,
        submitUserLifestyleResponse: false,
        submittingUserLifestyleResponse: false,
      },
      user: {
        profile: { preferredLocale: 'zh-HK' },
      },
    };
    const lifestyleForm = renderForTest(
      <LifestyleForm
        route={{
          params: { scrollToFutureMe: false },
        }}
      />,
      {
        initialState: initialStateWithFormRendered,
      },
    ).toJSON();

    expect(lifestyleForm).toMatchSnapshot();
  });

  it('should pass choices questions to MyChoices component', () => {
    const lifestyleForm = renderForTest(
      <LifestyleForm
        route={{
          params: { scrollToFutureMe: false },
        }}
      />,
      {
        initialState: {
          ...stateWithoutError,
          user: {
            profile: { preferredLocale: 'en-HK' },
          },
        },
      },
    );
    const myChoices = lifestyleForm.queryByType(MyChoices);

    expect(myChoices.props.questions).toEqual(
      english_questions.questionGroups[QuestionGroups.CHOICES],
    );
  });

  it('should navigate to health page when user submits lifestyle data', async () => {
    const navigate = jest.fn();
    const lifestyleForm = renderForTest(
      <LifestyleForm
        route={{
          params: { scrollToFutureMe: false },
        }}
        navigation={{ navigate }}
      />,
      {
        initialState: stateWithoutError,
      },
    );

    const button = lifestyleForm.getByType(Button);
    fireEvent.press(button);

    await flushMicrotasksQueue();

    expect(Alert.alert.mock.calls.length).toBe(0);
    expect(navigate.mock.calls.length).toBe(1);
    expect(navigate.mock.calls[0][0]).toBe(HEALTH);
  });

  it('should navigate to lifestyle form when server returns error', async () => {
    const navigate = jest.fn();

    const lifestyleForm = renderForTest(
      <LifestyleForm
        route={{
          params: { scrollToFutureMe: false },
        }}
        navigation={{ navigate }}
      />,
      {
        initialState: stateWithError,
      },
    );

    const button = lifestyleForm.getByType(Button);

    fireEvent.press(button);
    await flushMicrotasksQueue();
    jest.runOnlyPendingTimers();

    expect(Alert.alert.mock.calls.length).toBe(1);
  });

  it('should show a loader while submitting the lifestyle response', () => {
    const lifestyleForm = renderForTest(
      <LifestyleForm
        route={{
          params: { scrollToFutureMe: false },
        }}
      />,
      {
        initialState: stateWithSubmitResponseLoader,
      },
    );
    const loader = lifestyleForm.getByType(Loader);
    expect(loader).toBeDefined();
  });

  it('should render scroll view for sticky button to avoid scroll issues', () => {
    const lifestyleForm = renderForTest(
      <LifestyleForm
        route={{
          params: { scrollToFutureMe: false },
        }}
      />,
      {
        initialState: stateWithoutError,
      },
    );

    expect(
      lifestyleForm.queryAllByType(ScrollViewForStickyButton),
    ).toHaveLength(1);
  });

  describe('FutureMe', () => {
    let futureMe, initialState;
    beforeEach(() => {
      initialState = {
        ...stateWithoutError,
        health: {
          ...stateWithoutError.health,
          faceAging: {
            image: {
              uri: 'data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAASABIAAD',
            },
          },
        },
        user: {
          profile: { preferredLocale: 'en-HK' },
        },
      };
      const lifestyleForm = renderLifestyleForm({ initialState });
      futureMe = lifestyleForm.queryAllByType(FutureMe);
    });

    it('should render FutureMe component', () => {
      expect(futureMe).toHaveLength(1);
    });

    it('should pass change as prop to FutureMe', () => {
      expect(futureMe[0].props.change).toEqual(expect.any(Function));
    });

    it('should pass face aging image as prop to FutureMe', () => {
      const result = futureMe[0].props.initialValues;

      expect(result).toEqual({
        image: initialState.health.faceAging.image,
      });
    });
  });

  describe('on mount', () => {
    it('should download face aging photo', () => {
      const downloadFaceAgingPhoto = jest.fn();
      const intl = { formatMessage: jest.fn() };
      renderForTest(
        <PureLifestyleForm
          route={{
            params: { scrollToFutureMe: false },
          }}
          intl={intl}
          submitUserLifestyleResponse={jest.fn()}
          resetLoader={jest.fn()}
          submittingUserLifestyleResponse={false}
          fetchUserLifestyleResponse={jest.fn()}
          fetchUserLifestyleResponseEnd={jest.fn()}
          downloadFaceAgingPhoto={downloadFaceAgingPhoto}
          handleSubmit={jest.fn()}
        />,
      );

      expect(downloadFaceAgingPhoto).toHaveBeenCalledTimes(1);
    });
  });

  describe('on unmount', () => {
    it('should reset loader', () => {
      const resetLoader = jest.fn();
      const intl = { formatMessage: jest.fn() };
      const { unmount } = renderForTest(
        <PureLifestyleForm
          route={{
            params: { scrollToFutureMe: false },
          }}
          intl={intl}
          submitUserLifestyleResponse={jest.fn()}
          resetLoader={resetLoader}
          submittingUserLifestyleResponse={false}
          fetchUserLifestyleResponse={jest.fn()}
          fetchUserLifestyleResponseEnd={jest.fn()}
          downloadFaceAgingPhoto={jest.fn()}
          handleSubmit={jest.fn()}
        />,
      );

      unmount();

      expect(resetLoader).toHaveBeenCalledWith();
    });
  });

  describe('Dependent questions', () => {
    it('should submit inactive dependent values as null when not activated', async () => {
      const lifestyleForm = renderForTest(
        <LifestyleForm
          route={{
            params: { scrollToFutureMe: false },
          }}
        />,
        {
          initialState: {
            ...stateWithoutError,
            form: {
              lifestyleForm: {
                values: {
                  alcoholConsumptionFrequency: 'Never',
                  alcoholConsumptionAmount: 'MonthlyOrLess',
                },
              },
            },
          },
        },
      );

      fireEvent.press(lifestyleForm.queryAllByType(Button)[0]);
      await flushMicrotasksQueue();

      expect(submitUserLifestyleResponse).toHaveBeenCalledWith({
        alcoholConsumptionFrequency: 'Never',
        alcoholConsumptionAmount: null,
        avoidingTemptingFood: null,
        exerciseMinuteFrequency: null,
      });
    });
  });

  describe('MyHealth', () => {
    let lifestyleForm, myHealth;

    beforeEach(() => {
      lifestyleForm = renderForTest(
        <LifestyleForm
          route={{
            params: { scrollToFutureMe: false },
          }}
        />,
        {
          initialState: {
            ...stateWithoutError,
            user: {
              profile: { preferredLocale: 'en-HK' },
            },
          },
        },
      );
      myHealth = lifestyleForm.queryAllByType(MyHealth);
    });

    it('should render MyHealth section', () => {
      expect(myHealth).toHaveLength(1);
    });

    it('should pass health questions to MyHealth section', () => {
      expect(myHealth[0].props.questions).toEqual(
        english_questions.questionGroups[QuestionGroups.HEALTH],
      );
    });
  });

  describe('mapStateToProps', () => {
    const getInitialState = ({ healthData, fetchCompleted }) => ({
      form: {
        lifestyleForm: {
          values: {},
        },
      },
      health: {
        data: healthData,
        faceAging: { image: { uri: '' } },
        fetchFaceAgingImageCompleted: true,
        fetchUserLifestyleResponseCompleted: fetchCompleted,
        submittingUserLifestyleResponse: false,
      },
      user: {
        profile: { preferredLocale: 'en-HK' },
      },
    });

    const defaultValues = getInitialValuesFromQuestions(english_questions);

    it(`should use default value of each question as initialValues
    when fetching previous response is yet to complete`, () => {
      const initialState = getInitialState({
        healthData: {},
        fetchCompleted: false,
      });

      const { queryByType } = renderLifestyleForm({ initialState });
      const lifestyleForm = queryByType(PureLifestyleForm);

      const {
        image,
        ...initialValuesForQuestions
      } = lifestyleForm.props.initialValues;
      expect(initialValuesForQuestions).toEqual(defaultValues);
    });

    it(`should override questions' default values with previous response
    and use the result as initialValues
    when fetching previous response is completed`, () => {
      const previousResponse = {
        alcoholConsumptionFrequency: 'value1',
        highBloodPressure: 'value2',
      };
      const initialState = getInitialState({
        healthData: previousResponse,
        fetchFaceAgingImageCompleted: true,
        fetchCompleted: true,
      });

      const { queryByType } = renderLifestyleForm({ initialState });
      const lifestyleForm = queryByType(PureLifestyleForm);

      Object.keys(previousResponse).forEach(responseKey => {
        expect(lifestyleForm.props.initialValues[responseKey]).toEqual(
          previousResponse[responseKey],
        );
      });
    });

    it(`should use default value of exercise frequency
    if previous response has no value
    when fetching previous response is completed`, () => {
      const initialState = getInitialState({
        healthData: {},
        fetchCompleted: true,
      });

      const { queryByType } = renderLifestyleForm({ initialState });
      const lifestyleForm = queryByType(PureLifestyleForm);

      expect(lifestyleForm.props.initialValues.exerciseFrequency).toEqual('');
    });

    it(`should only override default value of exercise frequency
    if previous response has a value
    when fetching previous response is completed`, () => {
      const previousResponse = {
        exerciseFrequency: '2',
      };
      const initialState = getInitialState({
        healthData: previousResponse,
        fetchFaceAgingImageCompleted: true,
        fetchCompleted: true,
      });

      const { queryByType } = renderLifestyleForm({ initialState });
      const lifestyleForm = queryByType(PureLifestyleForm);

      expect(lifestyleForm.props.initialValues.exerciseFrequency).toEqual('2');
    });

    it('should render a disclaimer message', () => {
      const lifestyleForm = renderForTest(
        <LifestyleForm
          route={{
            params: { scrollToFutureMe: false },
          }}
        />,
        {
          initialState: {
            ...stateWithoutError,
            user: {
              profile: { preferredLocale: 'en-HK' },
            },
          },
        },
      );

      const disclaimerMessage = lifestyleForm.queryAllByText(
        messages['health.lifestyleForm.disclaimerMessage'],
      );

      expect(disclaimerMessage.length).toBe(1);
    });

    it('should render a disclaimer title', () => {
      const lifestyleForm = renderForTest(
        <LifestyleForm
          route={{
            params: { scrollToFutureMe: false },
          }}
        />,
        {
          initialState: {
            ...stateWithoutError,
            user: {
              profile: { preferredLocale: 'en-HK' },
            },
          },
        },
      );

      const disclaimerTitle = lifestyleForm.queryAllByText(
        messages['health.lifestyleForm.disclaimerTitle'],
      );

      expect(disclaimerTitle.length).toBe(1);
    });
  });
});
