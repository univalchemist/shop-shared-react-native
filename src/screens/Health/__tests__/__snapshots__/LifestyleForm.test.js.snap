// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LifestyleForm should show the lifestyle form page in chinese 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              enabled={false}
              hitSlop={
                Object {
                  "right": -700,
                }
              }
              maxDeltaY={20}
              minOffsetX={5}
              onHandlerStateChange={[Function]}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 1,
                            "justifyContent": "center",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "flexDirection": "row",
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "#9a9a9a",
                                "fontSize": 128,
                                "letterSpacing": -20,
                                "marginTop": -20,
                                "opacity": 0.4,
                              }
                            }
                          >
                             
                            .
                          </Text>
                          <Text
                            style={
                              Object {
                                "color": "#9a9a9a",
                                "fontSize": 128,
                                "letterSpacing": -20,
                                "marginTop": -20,
                                "opacity": 0.4,
                              }
                            }
                          >
                             
                            .
                          </Text>
                          <Text
                            style={
                              Object {
                                "color": "#9a9a9a",
                                "fontSize": 128,
                                "letterSpacing": -20,
                                "marginTop": -20,
                                "opacity": 0.4,
                              }
                            }
                          >
                             
                            .
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;

exports[`LifestyleForm should show the lifestyle form page in english 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              enabled={false}
              hitSlop={
                Object {
                  "right": -700,
                }
              }
              maxDeltaY={20}
              minOffsetX={5}
              onHandlerStateChange={[Function]}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 1,
                            "justifyContent": "center",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "flexDirection": "row",
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "#9a9a9a",
                                "fontSize": 128,
                                "letterSpacing": -20,
                                "marginTop": -20,
                                "opacity": 0.4,
                              }
                            }
                          >
                             
                            .
                          </Text>
                          <Text
                            style={
                              Object {
                                "color": "#9a9a9a",
                                "fontSize": 128,
                                "letterSpacing": -20,
                                "marginTop": -20,
                                "opacity": 0.4,
                              }
                            }
                          >
                             
                            .
                          </Text>
                          <Text
                            style={
                              Object {
                                "color": "#9a9a9a",
                                "fontSize": 128,
                                "letterSpacing": -20,
                                "marginTop": -20,
                                "opacity": 0.4,
                              }
                            }
                          >
                             
                            .
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
